### ISSUE TITLE
next/2 throws a FunctionClauseError when state.result is nil 

### ISSUE THREAD

**Issue by emarchak** [Omerlo-Technologies/ex_cycle#23]
Hi again!

Ran into another bug with this when creating schedules that have long intervals. I was able to recreate it with this test:

Repo:
```elixir
    test "occurrences have a long interval" do
      [datetime] =
        ExCycle.new()
        |> ExCycle.add_rule(
          :weekly,
          interval: 4,
          days: [:monday],
          start: ~D[2024-09-01]
        )
        |> ExCycle.occurrences(~U[2024-09-29 00:00:00Z])
        |> Enum.take(1)

      assert datetime == ~N[2024-09-30 00:00:00]
    end
```

Results in the following error:

```elixir
 ** (FunctionClauseError) no function clause matching in Date.beginning_of_week/2

     The following arguments were given to Date.beginning_of_week/2:

         # 1
         nil

         # 2
         :default

     Attempted function clauses (showing 2 out of 2):

         def beginning_of_week(%{calendar: Calendar.ISO} = date, starting_on)
         def beginning_of_week(%{calendar: calendar} = date, starting_on)

     code: |> ExCycle.occurrences(~U[2024-09-29 00:00:00Z])
     stacktrace:
       (elixir 1.17.1) lib/calendar/date.ex:921: Date.beginning_of_week/2
       (ex_cycle 0.5.8) lib/ex_cycle/validations/interval.ex:104: ExCycle.Validations.Interval.next/2
```

It looks like the validation is missing a nil check, but there might be a more elegant solution. Let me know if it's easiest for me to toss a PR up for you!

**Comment by Matsa59** [comment]
Hello, hum interesting problem. Is it critical for you?

**Comment by emarchak** [comment]
Yeah, we run into it pretty frequently with our rrules, it can get kinda whacky with our users. 

I can toss up a PR for you to review, if that's easiest! Happy to add a nil check or whatever approach you prefer.

**Comment by Matsa59** [comment]
Ok so I think we have to consider this as 2 separates problems. 

The first one is : weekly option doesn't work as expected
The second is : we MUST have an option to configure the week starts_at to be able to use `Date.day_of_week(~D[2016-10-31], :sunday)`

**Comment by Matsa59** [comment]
```elixir
ExCycle.new()
|> ExCycle.add_rule(
  :weekly,
  interval: 4,
  days: [:monday],
  start: ~D[2024-09-01],
  )
  |> ExCycle.occurrences(~U[2024-09-29 00:00:00Z], week_starting_on: :sunday)
  |> Enum.take(1)
```

this produce `[~N[2024-09-30 00:00:00]]` so I guess it's a fiable solution.

Without that you'll have `[~N[2024-10-21 00:00:00]]`

Is it good for you?

**Comment by Matsa59** [comment]
This should be fixed in version 0.6.0.
Thanks for your time
